braindecode/datasets/ecog_bci_competition.py

```diff
@@ -0,0 +1,113 @@
+import mne
+import numpy as np
+import scipy.io
+
+from braindecode.datasets import BaseDataset, BaseConcatDataset
+
+
+class EcogBCICompetition4(BaseConcatDataset):
+    """BCI competition IV ECoG dataset with finger movements.
+
+    Contains ECoG recordings for three patients moving fingers during the experiment.
+    Targets correspond to the time courses of the flexion of each of five fingers.
+    See http://www.bbci.de/competition/iv/desc_4.pdf for the dataset description.
+
+    Test labels can be downloaded from:
+    http://www.bbci.de/competition/iv/results/ds4/true_labels.zip
+
+    Data can be downloaded from (Dataset 4):
+    http://www.bbci.de/competition/iv/#dataset4
+
+    Notes
+    -----
+    When using this dataset in publications please cite [1]_ .
+
+    Parameters
+    ----------
+    path : str
+        Path to the folder with BCI competition IV dataset 4. All .mat files are
+        expected to be in this directory.
+    subject_ids: list(int) | int | None
+        (list of) int of subject(s) to be loaded. If None, load all available
+        subjects. Should be in range 1-3.
+
+    References
+    ----------
+    .. [1] Schalk, G., Kubanek, J., Miller, K.J., Anderson, N.R., Leuthardt, E.C.,
+        Ojemann, J.G., Limbrick, D., Moran, D.W., Gerhardt, L.A., and Wolpaw, J.R.
+        Decoding Two Dimensional Movement Trajectories Using Electrocorticographic Signals
+        in Humans, J Neural Eng, 4: 264-275, 2007.
+    """
+    possible_subjects = [1, 2, 3]
+
+    def __init__(self, path, subject_ids=None):
```

Comment:
@robintibor @agramfort For now to create this dataset user must specify path to the data (folder with .mat files downloaded from BCI IV competition site). To download the data you have to sign in with your email and password received in an email. Do you see any easier option to do that (without additional steps outside braindecode)? Do we want to create any specific path where we can put this kind of data (like mne_data)?

Reply:
so this dataset is not in moabb? I mean if you have to manually download anyways I feel supplying path seems fine to me...

Reply:
This dataset is not in moabb, so probably we will keep the dataset path as a parameter

Reply:
This dataset will be stored in `mne_data` as MOABB datasets, so problem solved :)